src/lib.rs

```diff
@@ -226,6 +226,13 @@ impl<T> Inner<T> {
             self.send_count -= diff;
         }
     }
+
+    /// Close the channel if there aren't any receivers present anymore
+    fn close_channel_if_no_receiver(&mut self) {
```

Comment:
Actually I don't think method name needs to be so descriptive.

```suggestion
    fn close_channel(&mut self) {
```

Reply:
must admit though that it feels a bit misleading, because it doesn't close it necessarily

Reply:
it's ok. It's internal only and the doc comment explains what it does.